torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -71,9 +81,31 @@ def pre_grad_passes(gm, example_inputs):
     gm.graph.lint()
     gm.recompile()
 
+    # Check if there is still unbacked SymInts after dyn_rewritten_pass.
+    # Trigger RestartAnalysis to graph break since Inductor can't handle them.
+    if has_unbacked_symint(gm, example_inputs):
+        raise RuntimeError("There is unbacked SymInt in the graph")
+
     return gm
 
 
+def has_unbacked_symint(gm, example_inputs):
+    fake_mode = detect_fake_mode(example_inputs)
+    for node in gm.graph.nodes:
+        if (
+            isinstance(node, torch.fx.node.Node)
+            and "example_value" in node.meta
+            and isinstance(node.meta["example_value"], FakeTensor)
+        ):
+            for x in node.meta["example_value"].shape:
```

Comment:
This is technically incomplete as you must check stride and storage offset. Instead, use the free_symbols function in symbolic_shapes to get all the symbols from a Tensor